scripts/contributions/get_results.py

```diff
@@ -0,0 +1,736 @@
+all the data after the replay is completed).
+
+I plan on adding more to this file, specifically the ability to graph much more
+data and get more useful statistics.
+
+If you have any suggestions/questions let me know :) - @Isaac
+'''
+
+pltInstalled = False
+
+try:
+	import os
+	import sys
+	import json
+	import glob
+	import math
+	import argparse
+except ImportError as e:
+	sys.stderr.write("WARNING: Module not found, full error:\n\n")
+	sys.stderr.write(e)
+
+try:
+	import matplotlib.pyplot as plt
+	pltInstalled = True
+except ImportError:
+	try:
+		usrIn = input('Matplotlib not found.\nWould you like this program to try and install matplotlib? (y/n) ')
+		if usrIn.lower() == 'y' or usrIn.lower() == 'yes':
+			import subprocess
+			subprocess.run(['python', '-m', 'pip', 'install', 'matplotlib'])
+
+			import matplotlib.pyplot as plt
+			pltInstalled = True
+			sys.stderr.write("\n\n")
+	except:
+		pltInstalled = False
+
+# handles all the arguments
+def parse_args():
+	ap = argparse.ArgumentParser(add_help=False, formatter_class=argparse.RawTextHelpFormatter)
+	ap.add_argument('-h', '--help', action='help', help='show this help message and exit\n\n')
+	ap.add_argument(
+		"-n", "--num",
+		default=1,
+		help="number of files (in order of date created) to analyze\n\n")
+	ap.add_argument(
+		"-a", "--all",
+		action='store_true',
+		help="runs every replay file in replay folder\n\n")
+	ap.add_argument(
+		"-v", "--verbose",
+		action='store_true',
+		help="will force the program to run each replay seperately and print information individual games\n\n")
+	ap.add_argument(
+		"-avg", "--averages",
+		nargs="*",
+		default=[],
+		help="data you would like the average of (not very useful right now)\nValid Options:\n\t- health\n\t- bits\n\t- cores\n\t- cores_spent\n\t- bits_spent\n\t- cores_on_board\n\n")
+	ap.add_argument(
+		"-f", "--file",
+		nargs="*",
+		default=[],
+		help="specify a replay file (or multiple) you'd like to analyze\n\n")
+	ap.add_argument(
+		"-g", "--graph",
+		nargs="*",
+		default=[],
+		help="specify what data you would like to be graphed - you must have matplotlib installed\n\nValid Options For Single Game:\n\t- health\n\t- bits\n\t- cores\n\t- cores_spent\n\t- bits_spent\n\t- cores_on_board\n\nValid Options For Multiple Games:\n\t- wins\n\n")
+	return vars(ap.parse_args())
+
+
+class Graph:
+	fig, ax, arg = None, None, None
+	pos = (0,0)
+	empty_plots = []
+
+	verbose_options = ['health', 'bits', 'cores', 'cores_spent', 'bits_spent', 'cores_on_board']
+	summary_options = ['wins']
+
+	@staticmethod
+	def init(arg):
+		Graph.clear()
+
+		num_sub_plots = arg.count(':') + 1
+		r = math.floor(math.sqrt(num_sub_plots))
+		c = math.ceil(num_sub_plots/r)
+
+		fig_size = [6.4, 4.8]
+		max_x, max_y = 15, 9
+		fig_size = [fig_size[0] * c, fig_size[1] * r]
+		if fig_size[0] > max_x: fig_size[0] = max_x
+		if fig_size[1] > max_y: fig_size[1] = max_y
+		plt.rcParams["figure.figsize"] = fig_size
```

Comment:
Not sure about this one.

Reply:
`rcParams` is specific to matplotlib and cannot be changed.